src/Data/Lens/At.purs

```diff
@@ -0,0 +1,45 @@
+module Data.Lens.At
+  ( At
+  , at
+  ) where
+
+import Prelude
+
+import Data.Identity (runIdentity, Identity(..))
+import Data.Map as M
+import Data.Maybe (Maybe(..), maybe)
+import Data.Set as S
+import Data.StrMap as SM
+import Data.Tuple
+
+import Data.Lens (LensP(), lens)
+import Data.Lens.Index (Index)
+
+class (Index m a b) <= At m a b where
```

Comment:
That's definitely an option... But I have a feeling that it might make it more difficult to use for monomorphic containers, like `StrMap`, etc. These can of course always be encoded using `Const` or something, but I have a feeling the "fully-applied" approach that we have now may be more modular & easier to work with.

I think the difference between the two approaches is essentially analogous to choosing between parameterization and fibration; these are equivalent (by the inverse image in one direction, and by the Grothendieck construction in the other), but sometimes fibration just works out a bit better.

Anyway, I'm definitely open to trying it out the way you suggest; but I was curious if you had any thoughts on the trade-offs.